src/engine/Server.cpp

```diff
@@ -841,6 +857,57 @@ Awaitable<void> Server::processQuery(
   co_return;
 }
 
+// ____________________________________________________________________________
+Awaitable<void> Server::processUpdate(
+    const ad_utility::url_parser::ParamValueMap& params, const string& update,
+    ad_utility::Timer& requestTimer,
+    const ad_utility::httpUtils::HttpRequest auto& request, auto&& send,
+    TimeLimit timeLimit) {
+  auto [queryHub, messageSender] =
+      createMessageSender(queryHub_, request, update);
+
+  auto [cancellationHandle, cancelTimeoutOnDestruction] =
+      setupCancellationHandle(messageSender.getQueryId(), timeLimit);
+
+  auto [pinSubtrees, pinResult] = determineResultPinning(params);
+  LOG(INFO) << "Processing the following SPARQL update:"
+            << (pinResult ? " [pin result]" : "")
+            << (pinSubtrees ? " [pin subresults]" : "") << "\n"
+            << update << std::endl;
+  QueryExecutionContext qec(index_, &cache_, allocator_,
+                            sortPerformanceEstimator_, std::ref(messageSender),
+                            pinSubtrees, pinResult);
+  auto plannedQuery = co_await setupPlannedQuery(
+      params, update, qec, cancellationHandle, timeLimit, requestTimer);
+  auto qet = plannedQuery.queryExecutionTree_;
```

Comment:
This code is identically duplicated between the query and the update case (the only differense is "UPDATE" vs "QUERY" in the Log message.
Just make it a function that 
returns `unique_ptr<SomeStructThatcontainsStuffThatWeNeedToAccessAndKeepAlive>` and does all this common setup of message senders, cancellation handles, qets, and whatNotElse.

Reply:
I am struggling with the type of `cancelTimeoutOnDestruction` lambda. Can you help with that @joka921? If skip the first two statements.

What I tried so far:
- std::function - lambda is not copy constructible, because it captures
- std::move_only_function - is c++23
- function pointer - lambda captures
- auto in the struct - lambda captures
- obtaining the type through template magic was also difficult because the function is a private member function in the same class